test_security_compliance.py

```
@@ -0,0 +1,157 @@
+async def verify_ssl_certificate(domain):
+    """Verify SSL certificate validity and properties"""
+    try:
+        print("\n🔒 Verifying SSL Certificate...")
+        context = ssl.create_default_context()
+        with socket.create_connection((domain, 443)) as sock:
+            with context.wrap_socket(sock, server_hostname=domain) as ssock:
+                cert = ssock.getpeercert()
+                
+                # Check certificate properties
+                if not cert:
+                    raise SecurityTestError("No SSL certificate found")
```

Ruff (0.12.2)

Abstract `raise` to an inner function

(TRY301)